chore: update module plugin docs and readme (#3782)
web-infra-dev/modern.js: packages/module/plugin-module-target/README.md

# Deprecated Package

> This package has been deprecated in next major version and is no longer being maintained.

Reviewer: ```suggestion
This package will be deprecated in the next major release and will no longer be maintained.
```
